src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,12 +8,18 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS =
+            "Person addresses must be in the following format BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = "[^ ].*, [^ ].*, [^ ].*, [^ ].*";
 
-    public final String value;
     private boolean isPrivate;
+    public final String value;
```

Comment:
Is there a need for `value` to be kept?
Would this cause a duplication of data?

Perhaps the `toString()` of **Address** could be changed to include **Block**, **Street**, etc. Thus this would remove the need for `value`

Reply:
That works, too. But the trim function I employed would remove extra spaces from user input for each field, so the address.toString may give a different result from the user input if the value is a combination of all fields' value . I am not sure whether this is desirable.